riscv/insns/ld.h

```diff
@@ -1,2 +1,7 @@
-require_rv64;
-WRITE_RD(MMU.load<int64_t>(RS1 + insn.i_imm()));
+require((xlen == 64) || p->extension_enabled(EXT_ZILSD));
```

Comment:
@aswaterman I noticed that the require checking is inconsistent. I do not know the implications of it, so wanted to check what the best solution is. The alternative to this would be to check in each branch of the if statement. See e.g. sd.h.
The compressed instructions miss any require check in the XLEN != 32 branch. I assume this must be added there. Will do if I know your preference.

Reply:
I think the strategy you've used for `ld`, using a single `require` at the top, is clean.

For e.g. `c.ld`, it looks like I did screw up by not requiring RV64.  For those, I think we want

    require_extension(EXT_ZCA);
    require((xlen == 64) || p->extension_enabled(EXT_ZILSD));

Thanks for fixing it.

Reply:
fixed in 0ddd72a1